### PR TITLE
Campaigns can only be created by an Artist

### PR DESCRIPTION
**Pull request by ESteinman**
## PT-link:
https://www.pivotaltracker.com/story/show/160215745

## User story:
```
As a System owner, 
In order to eliminate fake and non-serious Campaigns
We would like the Campaigns to only be created by Artists.
```

## Tasks
- Gem `pundit`
- Add role:  `Artist` (enum)
- Restrict campaign creation to Artists only
- Authorisation: limit `Campaigns#create` to `Artist`
- Add `Associations` - `belongs_to :user` and `has_many :campaigns`

## New gems
- gem `pundit`
- gem `pundit-matchers`

## New Migrations
`AddRoleToUsers`
`AddCampaignsToUser`

## What've learned
- Add `@javascript`...
- `current_user&.artist` instead of `user_signed_in && current_user.artist`

### CODE REVIEW

File: spec/models/campaign_spec.rb [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
+  end
+  
   describe 'Factory' do
     it 'should have valid Factory' do
       expect(create(:campaign)).to be_valid 
     end
   end
 
+  describe 'Associations' do
+    it {is_expected.to belong_to :user}
```
Comment (magnus-thor) [inline spec/models/campaign_spec.rb]: Be consistent in usage of spaces after `{` and before `}`

File: features/campaign_can_only_be_created_by_artist.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Feature: A Campaign can only be created by an Artist
+    As a System owner, 
+    In order to eliminate fake and non-serious Campaigns
+    We would like the Campaigns to only be created by Artists.
+
+    Background:
+        Given the following users exist
+        | email             | role      |
+        | user@fan.se       | fan       |
+        | user@artist.se    | artist    |
```
Comment (magnus-thor) [inline features/campaign_can_only_be_created_by_artist.feature]: be consistent with spaces, look at next feature file how the table is written

File: features/campaign_can_only_be_created_by_artist.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    Scenario: Fan tries to access Create Campaign page via link
+        Given I am logged in as 'user@fan.se'
+        And I try to access 'Create Campaign' page
```
Comment (magnus-thor) [inline features/campaign_can_only_be_created_by_artist.feature]: try to use all of Given When Then
`When I try to access 'Create Campaign' page`

Comment (ESteinman) [reply features/campaign_can_only_be_created_by_artist.feature]: I understand what you mean, but I think the current wording makes more sense in this case.

**Pull request merged**